amr-wind/immersed_boundary/complex_terrain/complex_terrain_ops.H

```diff
@@ -87,6 +89,71 @@ struct ComputeForceOp<
     void operator()(typename GeomTrait::DataType&) {}
 };
 
+template <typename IbType>
+struct ComputeSourceOp<
+    IbType,
+    std::enable_if_t<std::is_base_of<ComplexTerrainType, IbType>::value>>
+{
+    void operator()(
+        typename IbType::DataType& data,
+        const int lev,
+        const amrex::MFIter& mfi,
+        const amrex::Geometry& geom)
+    {
+        Field& source = data.sim().repo().get_field("ib_src_term");
+        const Field& level = data.sim().repo().get_field("ib_levelset");
+        const Field& normal = data.sim().repo().get_field("ib_normal");
+        const Field& velocity = data.sim().repo().get_field("velocity");
+        const Field& density = data.sim().repo().get_field("density");
+
+        const auto& bx = mfi.tilebox();
+        const auto& varr = velocity(lev).array(mfi);
+        const auto& darr = density(lev).array(mfi);
+        const auto& phi_arr = level(lev).array(mfi);
+        const auto& norm_arr = normal(lev).array(mfi);
+        auto sarr = source(lev).array(mfi);
+
+        const amrex::Real cd = data.meta().drag_coefficient;
+        const amrex::Real eps = data.meta().eps_spread;
+
+        amrex::ParallelFor(
+            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                // A lot of this is unnecessary for a flat plate, but is
+                // general enough that it could be broken off into its own
+                // function at some point.
+                const amrex::Vector<amrex::Real> vel = {
+                    {varr(i, j, k, 0), varr(i, j, k, 1), varr(i, j, k, 2)}};
+                const amrex::Real vmag = utils::vec_mag(vel.data());
+                const amrex::Vector<amrex::Real> norm = {
+                    {norm_arr(i, j, k, 0), norm_arr(i, j, k, 1),
+                     norm_arr(i, j, k, 2)}};
+                // Outside the band if the normal vector is assigned to be zero
+                const amrex::Real nmag = utils::vec_mag(norm.data());
+                if (nmag > 0.5) {
```

Comment:
If we leave out the nmag check,
in the case where nmag = 0, vtan = v - (v.n) n = v, which is not what we want.
We don't want to compute a drag force when nmag = 0.

Regarding 0.5, it was just a safe value for comparison.  It should only be zero or one, but it could accidentally be macheps or 1-macheps, so I wanted to avoid comparing with those.  0.5 was just nice and safe, but I could see where it could be confusing to the reader.